pkg/monitors/diskio/diskio_others.go

```diff
@@ -82,10 +77,9 @@ func (m *Monitor) Configure(conf *Config) error {
 		return err
 	}
 
-	// gather metrics on the specified interval
-	utils.RunOnInterval(ctx, func() {
-		m.emitDatapoints()
-	}, time.Duration(conf.IntervalSeconds)*time.Second)
-
 	return nil
 }
+
+func (m *Monitor) Collect(ctx context.Context) {
```

Comment:
Add an interface type assertion in this module (`var _ Collectable = new(Monitor)`)